cmd/dvotenode/dvotenode.go

```diff
@@ -545,6 +545,27 @@ func main() {
 			log.Fatal(err)
 		}
 
+		w3uris := []string{}
+		for idx, web3Endpoint := range globalCfg.W3Config.W3External {
+			web3EndpointTrimmed := strings.Trim(web3Endpoint, `"[]`)
+			log.Debugf("web3endpoint %d: %s", idx, web3EndpointTrimmed)
+			switch {
+			case strings.HasPrefix(web3EndpointTrimmed, "ws"):
+				w3uris = append(w3uris, web3EndpointTrimmed)
+			case strings.HasSuffix(web3EndpointTrimmed, "ipc"):
+				w3uris = append(w3uris, web3EndpointTrimmed)
+			default:
+				log.Warnf(`invalid web3 endpoint %s must be websocket or IPC
+						for event subscription`, web3EndpointTrimmed)
+			}
+		}
+
+		if len(w3uris) == 0 {
+			log.Fatal(`no web3 endpoint found, oracle needs at least
+			one valid web3 endpoint to handle Ethereum contract events`)
```

Comment:
FYI this log line will have a newline and lots of whitespace. I'd just join it as a single line, but if you really want to split it:

```
log.Fatal("foo bar " + 
    "baz")
```